Sources/ProjectDescription/Template.swift

```diff
@@ -0,0 +1,76 @@
+public struct Template: Codable {
+    public let description: String
+    public let attributes: [Attribute]
+    public let files: [File]
+    public let directories: [String]
+    
+    public init(description: String,
+                arguments: [Attribute] = [],
+                files: [File] = [],
+                directories: [String] = []) {
+        self.description = description
+        self.attributes = arguments
+        self.files = files
+        self.directories = directories
+        dumpIfNeeded(self)
+    }
+}
+
+public extension String.StringInterpolation {
+    mutating func appendInterpolation(_ value: Attribute) {
+        switch value {
+        case let .required(name), let .optional(name, default: _):
+            appendInterpolation("{{ \(name) }}")
+        }
+    }
+}
+
+public struct File: Codable {
+    public let path: String
+    public let contents: String
+    
+    public init(path: String, contents: String) {
+        self.path = path
+        self.contents = contents
+    }
+}
+
+public enum Attribute: Codable {
```

Comment:
Since this is specific to the `Template` entity, what about placing it under it? E.g.: `Template.File`

Reply:
Yeah, that's a good idea, will do 👍